### PR TITLE
Rfc 1 draft wip

### PR DESCRIPTION
**Pull request by andygotz**
Added some basic description. Feedback welcomed!

### CODE REVIEW

File: 1/README.md [hunk 1]
```diff
@@ -3,16 +3,17 @@ domain: rfc.tango-controls.org
...
 
-This document describes Tango, a control system framework. This RFC define the specification of a control system following the philosophy of Tango.
+This document describes Tango, a control system framework. This RFC define the specification of a control system implementing the philosophy of Tango.
 
 See also: Y/OtherTemplate
 
 ## Preamble
 
-Copyright (c) 2019 MAX IV Laboratory.
+Copyright (c) 2019 European Synchrotron (ESRF).
```
Comment (lorenzopivetta) [inline 1/README.md]: Shouldn't this read "Copyright (c) 2019 Tango Controls" ?

File: 1/README.md [hunk 2]
```diff
@@ -21,16 +22,36 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 
 ## Tango Specification
-The TANGO control system is a device-oriented controls toolkit for controlling any kind of hardware or software and building SCADA systems.
-
+The TANGO control system is a toolkit for building distributed control systems which are device oriented.
```
Comment (lorenzopivetta) [inline 1/README.md]: Likely, this going to be the very first section one would read when coming to Tango RFC... suggest to explain a little bit the concept "device oriented", e.g. "... which are device oriented, modelling the hardware equipment or the software component into the device abstraction". I think this can be improved how it is stated, though.

File: 1/README.md [hunk 3]
```diff
@@ -21,16 +22,36 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
 
 ## Tango Specification
-The TANGO control system is a device-oriented controls toolkit for controlling any kind of hardware or software and building SCADA systems.
-
+The TANGO control system is a toolkit for building distributed control systems which are device oriented. 
 
 ### Goals
 
-Tango aims to 
+Tango aims to be extremely flexible and can be used to interface any kind of hardware or software system which can be accessed via one of the software languages supported by Tango (C++, Python, Java and JavaScript).
```
Comment (lorenzopivetta) [inline 1/README.md]: s/accessed/interfaced/

File: 1/README.md [hunk 4]
```diff
@@ -21,16 +22,36 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
+
+The goal of Tango is to provide a way of representing all physical and software systems which need to be controlled by a control system as Tango Devices.
+
+Devices are software components which have Commands (methods), Attributes (data) and Properties (configuration) and are accessible on the network or locally.
+
+A Tango control system consists of a collection of Tango Devices, an (optional) Database and a set of applications for configuring and controlling the Devices via well defined Application Programmers Interfaces (APIs). Tango defines and maintains official APIs for C++, Python, Java, JavaScript and REST.
```
Comment (lorenzopivetta) [inline 1/README.md]: s/Application Programmers Interface/Application Programming Interface/
Also, not sure to state "(optional) Database" here... I'm in the idea Tango requires the Database, and working without is just a workaround for limited setups and use cases...

File: 1/README.md [hunk 5]
```diff
@@ -21,16 +22,36 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
+
+The Tango Database serves the purpose of defining and declaring all Devices in the control system and their configuration parameters. The Database also stores the network address of each Device and serves as Naming Service for clients who need to establish a connection to any or all Devices.
```
Comment (lorenzopivetta) [inline 1/README.md]: I'd remove "stores the network addresses" as it is likely to me misinterpreted here... rephrase as "The Database also serves as Naming Service for clients who need to establish a connection to any or all Devices".

File: 1/README.md [hunk 6]
```diff
@@ -21,16 +22,36 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
+The Tango toolkit was mainly developed for research facilities needs, but the idea and concept (philosophy) behind was to create a framework for control systems in general. The concept is based on the idea that the concept of an object or component is ideally suited to designing and implementing control systems. TANGO was based on this concept and can be described as an object oriented control system. A number of concepts were defined as being necessary to implement a TANGO control system (ref "TANGO - AN OBJECT ORIENTED CONTROL SYSTEM BASED ON CORBA" by Chaize et. al., ICALEPCS 1999). These are:
+
+1. Device network object with the minimum of functionality necessary for a distributed control object
+2. Use CORBA communication protocol while hiding details
+3. Support following communication paradigms - synchronous, asynchronous and event driven
+4. Support the implementation of new devices by sub-classing Device and by composing new types of Devices from existing types of Devices
+5. Support modern programming patterns like multi-threading
+6. A naming database which stores configuration information
+7. A generic set of Data Types
```
Comment (lorenzopivetta) [inline 1/README.md]: Suggest to rephrase "A generic, but limited, set of Data Types"

File: 1/README.md [hunk 7]
```diff
@@ -21,16 +22,36 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
 
 The software used in large facilities can be compared with the software for stock markets – huge amount of data must be displayed on the monitor in a real time and being processed and being saved in databases for post processing.
 
-So, the toolkit was mainly developed for research facilities needs, but the idea and concept (philosophy) behind was to create a framework.
+The Tango toolkit was mainly developed for research facilities needs, but the idea and concept (philosophy) behind was to create a framework for control systems in general. The concept is based on the idea that the concept of an object or component is ideally suited to designing and implementing control systems. TANGO was based on this concept and can be described as an object oriented control system. A number of concepts were defined as being necessary to implement a TANGO control system (ref "TANGO - AN OBJECT ORIENTED CONTROL SYSTEM BASED ON CORBA" by Chaize et. al., ICALEPCS 1999). These are:
```
Comment (lorenzopivetta) [inline 1/README.md]: I'd rephrase this as:
"The Tango toolkit was mainly developed for research facilities needs, but the idea and concept (philosophy) behind was to create a framework for control systems in general. It is based on the idea that distributed objects are really suited to design and implement control systems. TANGO was based on this concept, and can be described as an Object Oriented distributed control system. A number of requirements were defined in the original design (ref... TANGO - An Object ...):

1. ...
2. ...

File: 1/README.md (hunk 1 above)
Comment (andygotz) [reply 1/README.md]: Right!

File: 1/README.md [hunk 8]
```diff
@@ -57,18 +57,16 @@ https://www.elettra.trieste.it/ICALEPCS99/proceedings/papers/wa2i01.pdf
 
 Additionally, it aims to:
 
-* Provide ... 
+* Provide an Open Source software toolkit for building distirbuted object oriented control systems. It should be freely available for all applications including industry. Tango Controls is sufficiently complete to control any small or large installation. Provide the necessary libraries for 3 programming languages which allow hardware to be interfaced and modelled in a device object model. Provide the tools for configuring, managing, monitoring and archiving large numbers of control points implemented as device attributes. Provide naming, browsing and hierarchies of devices and groups of devices.
```
Comment (bourtemb) [inline 1/README.md]:
```suggestion
* Provide an Open Source software toolkit for building distributed object oriented control systems. It should be freely available for all applications including industry. Tango Controls is sufficiently complete to control any small or large installation. Provide the necessary libraries for 3 programming languages which allow hardware to be interfaced and modeled in a device object model. Provide the tools for configuring, managing, monitoring and archiving large numbers of control points implemented as device attributes. Provide naming, browsing and hierarchies of devices and groups of devices.
```

Comment (bourtemb) [reply 1/README.md]: I suggested 2 typos fixes (distributed and modeled)... Not so easy to see with github interface it seems.
I think you can put 1 sentence per line, it will be easier to suggest changes. If you don't end up your lines with 2 spaces, it will result in the same presentation.

File: 1/README.md [hunk 9]
```diff
@@ -3,16 +3,17 @@ domain: rfc.tango-controls.org
 shortname: 1/TANGO
 name: Tango Control System
 status: raw
-editor: Vincent Hardion (vincent.hardion@maxiv.lu.se)
+editor: Lorenzo Pivetta (lorenzo.pivetta@elettra.eu)
+contributor: Andy Götz (andy.gotz@esrf.fr)
 ---
 
-This document describes Tango, a control system framework. This RFC define the specification of a control system following the philosophy of Tango.
+This document describes Tango, a control system framework. This RFC define the specification of a control system implementing the philosophy of Tango.
```
Comment (bourtemb) [inline 1/README.md]:
```suggestion
This document describes Tango, a control system framework. This RFC defines the specification of a control system implementing the philosophy of Tango.
```

File: 1/README.md [hunk 10]
```diff
@@ -21,32 +22,51 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
 
 ## Tango Specification
-The TANGO control system is a device-oriented controls toolkit for controlling any kind of hardware or software and building SCADA systems.
-
+The TANGO control system is a toolkit for building distributed control systems which are device oriented. 
 
 ### Goals
 
-Tango aims to 
+Tango aims to be extremely flexible and can be used to interface any kind of hardware or software system which can be accessed via one of the software languages supported by Tango (C++, Python, Java and JavaScript).
```
Comment (piogor) [inline 1/README.md]: I would somehow mark that the intention is to support other languages, too.

**Review by lorenzopivetta — APPROVED** [review]

File: 1/README.md [hunk 11]
```diff
@@ -21,32 +22,51 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
+
+The goal of Tango is to provide a way of representing all physical and software systems which need to be controlled by a control system as Tango Devices.
+
+Devices are software components which have Commands (methods), Attributes (data) and Properties (configuration) and are accessible on the network or locally.
+
+A Tango control system consists of a collection of Tango Devices, a Tango Database and a set of applications for configuring and controlling the Devices via well defined Application Programmers Interfaces (APIs). Tango defines and maintains official APIs for C++, Python, Java, JavaScript and REST.
+
+The Tango Database serves the purpose of defining and declaring all Devices in the control system and their configuration parameters. The Database also stores Naming Service for clients who need to establish a connection to any or all Devices.
```
Comment (piogor) [inline 1/README.md]:
```suggestion
The Tango Database serves the purpose of defining and declaring all Devices in the control system and their configuration parameters. The Database also stores Naming Service for clients who need to establish a connection to any or all Devices (see [RFC 6/Database](/6)).
```

File: 1/README.md [hunk 12]
```diff
@@ -21,32 +22,51 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
-The TANGO control system is a device-oriented controls toolkit for controlling any kind of hardware or software and building SCADA systems.
-
+The TANGO control system is a toolkit for building distributed control systems which are device oriented. 
 
 ### Goals
 
-Tango aims to 
+Tango aims to be extremely flexible and can be used to interface any kind of hardware or software system which can be accessed via one of the software languages supported by Tango (C++, Python, Java and JavaScript).
+
+The goal of Tango is to provide a way of representing all physical and software systems which need to be controlled by a control system as Tango Devices.
```
Comment (piogor) [inline 1/README.md]:
```suggestion
The goal of Tango is to provide a way of representing all physical and software systems which need to be controlled by a control system as Tango Devices (see [RFC 2/Device](/2)).
```

File: 1/README.md [hunk 13]
```diff
@@ -21,32 +22,51 @@ This Specification is a [free and open standard](http://www.digistan.org/open-st
...
+
+Devices are software components which have Commands (methods), Attributes (data) and Properties (configuration) and are accessible on the network or locally.
```
Comment (piogor) [inline 1/README.md]:
```suggestion
Devices are software components which have [Commands](/3) (methods), [Attributes](/4) (data) and [Properties](/5) (configuration) and are accessible on the network or locally.
```

Comment (piogor) [reply 1/README.md]: Should Pipes be listed here?

**Comment by bourtemb** [comment]
@andygotz , @lorenzopivetta , @piogor , I cancelled the merge and did a git pull --rebase instead of the previously done merge, so now you see only the commits related to this PR in the PR commits.

**Comment by andygotz** [comment]
@bourtemb mille mercis! Now we need to add some info on the architecture.

File: 1/README.md (hunk 1 above)
Comment (hardion) [reply 1/README.md]: Damm you discovered my secret plan!

**Comment by lorenzopivetta** [comment]
Thanks @bourtemb, looks much cleaner now. @bourtemb , @piogor , @andygotz we still have one pending comment (e.g. Pipes RFC number). I think we can merge now and fix it later... what do you think?
